src/averageAges.js

```diff
@@ -15,11 +15,30 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const peopleSexFilter = people.filter(x => x.sex === 'm');
+
+  if (century === undefined || typeof century !== 'number') {
+    const ages = peopleSexFilter.map(person => {
+      return person.died - person.born;
+    });
+    const everageAge = ages.reduce((sum, x) => sum + x);
+
+    return everageAge / ages.length;
+  } else {
+    const ageFilter = peopleSexFilter.filter((person) => {
+      const ageCounter = Math.ceil(person.died / 100);
+
+      return ageCounter === century;
+    });
+
+    const ages = ageFilter.map(person => {
+      return person.died - person.born;
+    });
+
+    const everageAge = ages.reduce((sum, x) => sum + x);
+
+    return everageAge / ages.length;
+  }
```

Comment:
Good, but you can take repeated logic out of if else statement

Reply:
agree make function for repeating logic